internal/pkg/fleetapi/action.go

```diff
@@ -348,6 +406,54 @@ func (a *ActionCancel) String() string {
 	return s.String()
 }
 
+func (a *ActionCancel) AckEvent() AckEvent {
+	return newAckEvent(a.ActionID, a.ActionType)
+}
+
+// ActionDiagnostics is a request to gather and upload a diagnostics bundle.
+type ActionDiagnostics struct {
+	ActionID   string `json:"action_id"`
+	ActionType string `json:"type"`
+	FileID     string `json:"-"`
+	Err        error  `json:"-"`
+}
+
+// ID returns the ID of the action.
+func (a *ActionDiagnostics) ID() string {
+	return a.ActionID
+}
+
+// Type returns the type of the action.
+func (a *ActionDiagnostics) Type() string {
+	return a.ActionType
+}
+
+func (a *ActionDiagnostics) String() string {
+	var s strings.Builder
+	s.WriteString("action_id: ")
+	s.WriteString(a.ActionID)
+	s.WriteString(", type: ")
+	s.WriteString(a.ActionType)
+	return s.String()
+}
+
+func (a *ActionDiagnostics) AckEvent() AckEvent {
+	event := newAckEvent(a.ActionID, a.ActionType)
+	if a.Err != nil {
+		event.Error = a.Err.Error()
+	}
+	if a.FileID != "" {
+		var data struct {
+			FileID string `json:"file_id"`
+		}
+		data.FileID = a.FileID
+		p, _ := json.Marshal(data)
+		event.Data = p
+	}
+
+	return event
+}
```

Comment:
@juliaElastic, the diagnostics action will now ack with `data.file_id: VALUE`.
The `data` field in the result doc is populated from the event https://github.com/elastic/fleet-server/blob/main/internal/pkg/api/handleAck.go#L165

we should not need to do any changes for fleet-server to handle the diagnostics requests